app/reducers/modal.js

```diff
@@ -0,0 +1,81 @@
+// ------------------------------------
+// Initial State
+// ------------------------------------
+const initialState = {
+  modals: []
+}
+
+// ------------------------------------
+// Constants
+// ------------------------------------
+export const OPEN_MODAL = 'OPEN_MODAL'
+export const CLOSE_MODAL = 'CLOSE_MODAL'
+
+/**
+ * Generates uniq id for modals
+ */
+const genId = () =>
```

Comment:
There is a similar code in notifications. Could you please extract that into an util and reuse ?

Reply:
Good call. Done.